src/source.cpp

```diff
@@ -367,9 +372,11 @@ void load_custom_source_library()
   // reset errors
   dlerror();
 
-  // get the function from the library
-  using sample_t = Particle::Bank (*)(uint64_t* seed);
-  custom_source_function = reinterpret_cast<sample_t>(dlsym(custom_source_library, "sample_source"));
+  // get the function to create the CustomSource from the library
+  create_custom_source_t* create_custom_source = (create_custom_source_t*) dlsym(custom_source_library, "openmc_create_source");
+
+  // create a pointer to an instance of the CustomSource
+  custom_source = create_custom_source(custom_source_parameters.c_str());
```

Comment:
If you go with the suggestion to use `std::string` directly, this becomes
```suggestion
  custom_source = create_custom_source(custom_source_parameters);
```